File: Easy_two_sum_number_target/two_sum_number.js
function twoNumberSum(array, targetSum) {
    // Write your code here.
    matchTable = {};

    for(const num of array)
    {
        const matchNumber = targetSum - num;
       
        if( matchNumber in matchTable)
        {
            return [matchNumber, num];
        }
        else 
        {
            matchTable[num] = true;
        }

    }

    return [];
  }

  array = [3, 5, -4, 8, 11, 1, -1, 6];
  targetSum = 10;

  console.log(twoNumberSum(array, targetSum));
  
 
 

File: Easy_two_sum_number_target/two_sum_number.py
def twoNumberSum(array, targetSum):
    matchTable = {}
    for num in array:
        if(targetSum - num in matchTable):
            return [num, targetSum-num]
        else:
            matchTable[num] = True
    return []

if __name__ == '__main__':
    array = [3, 5, -4, 8, 11, 1, -1, 6]
    targetSum = 10
    print(twoNumberSum(array, targetSum));